coredb/src/metric/time_series.rs

```
@@ -168,6 +166,22 @@ impl TimeSeries {
 
     retval
   }
+
+  // Function to check if given string is metric and return extracted metric name
+  pub fn is_metric(s: &str) -> bool {
+    s.starts_with(METRIC_NAME_PREFIX)
+  }
+
+  // Function to extract metric name from given string
+  pub fn extract_metric_name(s: &str) -> String {
+    s.split(LABEL_SEPARATOR).collect::<Vec<&str>>()[1].to_string()
```

nit: may want to return an error is 's' does not correspond to a metric name label, and may want to utilize extract_label

---

this check is already done by caller of this function. Are you suggesting combining it?